File: javascript/examples/smoke.mjs
import { AmanaUnionCargillsBankDocsClient } from "../client.mjs";

const client = new AmanaUnionCargillsBankDocsClient({ defaultDelayMs: 1000 });
console.log("smoke", AmanaUnionCargillsBankDocsClient.slug, "->", "amanaRates");
const data = await client.amanaRates();
const preview = typeof data === "string" ? data.slice(0, 200) : JSON.stringify(data)?.slice(0, 200);
console.log("ok", preview);

File: python/smoke.py
"""Quick client smoke — polite delay on."""

from __future__ import annotations

import json

from amana_union_cargills_bank_docs import AmanaUnionCargillsBankDocsClient


def json_preview(data: object, limit: int = 200) -> str:
    try:
        return json.dumps(data, ensure_ascii=False)[:limit]
    except TypeError:
        return str(data)[:limit]


with AmanaUnionCargillsBankDocsClient(default_delay_seconds=1.0) as client:
    print("smoke", client.slug, "->", "amana_rates")
    data = client.amana_rates()
    preview = data[:200] if isinstance(data, str) else json_preview(data)
    print("ok", preview)
